Checks/PWR079/README.md

```diff
@@ -0,0 +1,336 @@
+# PWR079: Avoid undefined behavior due to uninitialized variables
+
+### Issue
+
+Accessing an uninitialized variable can lead to undefined behavior due to its
+indeterminate value.
+
+### Actions
+
+Always initialize variables before using them, helping ensure deterministic
+behavior and prevent bugs in the code.
+
+### Relevance
+
+Some programming languages automatically initialize variables to default
+values, but Fortran, C, and C++ often do not. Consequently, variables left
+uninitialized contain unpredictable data until they are explicitly set by the
+programmer:
+
+- **Fortran:** Reading any variable that has not been explicitly initialized
+  results in undefined behavior.
+
+- **C:** Automatic (i.e., declared within functions) variables are not
+  initialized by default. However, `static`, `thread_local`, and file-scope
+  variables are zero-initialized (e.g., scalar values are set to `0`).
+
+- **C++:** Similar to C, with additional rules for default initialization of
+  class-like objects and array elements in certain contexts.
+
+Since uninitialized variables may contain any arbitrary values, reading and
+using them can lead to undefined behavior, potentially causing incorrect
+results, crashes, or other unintended outcomes. Compilers are not required to
+warn about these issues and, even if they do, they typically still allow the
+code to compile and run.
+
+Some compilers may appear to "help" by zero-initializing variables under
+certain conditions (e.g., specific compilation flags). While this can make
+technically incorrect code run as originally intended, relying on such
+incidental behavior creates a false sense of security and masks underlying
+logical errors. Ultimately, this can vanish under different compilation
+settings and can vary between compilers.
+
+Lastly, while some compilers provide options to automatically initialize
+certain data types (e.g., `gfortran`'s `-finit-integer=<value>` or `gcc`'s
+`-ftrivial-auto-var-init=<value>`), these features reduce portability to other
+development environments and hide problems in the code rather than addressing
+them.
+
+### Code examples
+
+#### C
+
+Consider the following code, which aims to sum the elements of an array:
+
+```c
+// example_array.c
+#include <stdio.h>
+
+__attribute__((pure)) double sum_array(double *array, int size) {
+  double sum;
+
+  for (int i = 0; i < size; ++i) {
+    sum += array[i];
+  }
+
+  return sum;
+}
+
+int main() {
+  double array[] = {0.24, 0.33, 0.17, 0.89, 0.05};
+  double sum = sum_array(array, 5);
+  printf("Sum is: %f\n", sum);
```

Comment:
Maybe avoid using `sum` here to remove ambiguity with the `sum` of the `sum_array` function (same in the Fortran example).

Reply:
Agreed.

```suggestion
printf("Sum is: %f\n", sum_array(array, 5));
```

Reply:
> to remove ambiguity

Good catch! Done in both the C and Fortran code examples.